File: 1957-delete-characters-to-make-fancy-string/1957. Delete Characters to Make Fancy String.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res = ""
        idx = 0
        while idx < len(s):
            count = 1
            curr = s[idx]
            temp_idx = idx
            while temp_idx+1 < len(s) and s[temp_idx+1] == curr:
                count += 1
                temp_idx += 1
            if count >= 3:
                res += curr + curr
                while idx < len(s) and s[idx] == curr:
                    idx += 1
            else:
                res += curr
                idx += 1
        return res
        

File: 1957-delete-characters-to-make-fancy-string/1957. Delete Characters to Make Fancy String.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res = "";
        res+=s[0];
        int curr = 1;
        for (int i = 1; i < s.size(); i++) {
            if (s[i] == s[i-1]) {
                curr++;
                if (curr < 3) {
                    res += s[i];
                }
            } else {
                curr = 1;
                res += s[i];
            }
        }
        return res;
    }
};